homeassistant/components/zwave/__init__.py

```diff
@@ -445,6 +459,32 @@ def set_config_parameter(service):
         _LOGGER.info("Setting config parameter %s on Node %s "
                      "with value %s and size=%s", param, node_id, value, size)
 
+    def add_association(service):
+        """Add a association in the zwave network."""
+        node_id = service.data.get(const.ATTR_NODE_ID)
+        target_node_id = service.data.get(const.ATTR_TARGET_NODE_ID)
+        group = service.data.get(const.ATTR_GROUP)
+        instance = service.data.get(const.ATTR_INSTANCE, 0x00)
+
+        node = ZWaveGroup(group, NETWORK, node_id)
+        node.add_association(target_node_id, instance)
+        _LOGGER.info("Adding association for node:%s in group:%s "
+                     "target node:%s, instance=%s", node_id, group,
+                     target_node_id, instance)
+
+    def remove_association(service):
```

Comment:
This is almost a copy of `add_association` service handler.  Only the method and log text differ. This would benefit from a refactor but that should probably look at all services in the module, so would be a different PR.